src/main/java/org/prebid/server/bidder/cpmstar/CPMStarBidder.java

```diff
@@ -0,0 +1,164 @@
+package org.prebid.server.bidder.cpmstar;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.Json;
+import org.apache.commons.collections4.CollectionUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.cpmstar.ExtImpCPMStar;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class CPMStarBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpCPMStar>> CPM_STAR_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpCPMStar>>() {
+            };
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    private final String endpointUrl;
+
+    public CPMStarBidder(String endpointUrl) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        try {
+            final BidRequest bidRequest = processRequest(request, errors);
+
+            return Result.of(Collections.singletonList(createSingleRequest(bidRequest)), errors);
+        } catch (IllegalArgumentException e) {
+            return Result.of(Collections.emptyList(),
+                    Collections.singletonList(BidderError.badInput(e.getMessage()))
+            );
+        }
+    }
+
+    private BidRequest processRequest(BidRequest bidRequest, List<BidderError> errors) {
+        if (CollectionUtils.isEmpty(bidRequest.getImp())) {
+            throw new IllegalArgumentException("No Imps in Bid Request");
+        }
+
+        final List<Imp> validImpList = new ArrayList<>();
+        for (final Imp imp : bidRequest.getImp()) {
+            try {
+                validImpList.add(parseAndValidateImp(imp));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
```

Comment:
In go, they do not collect all exception (but some bidder does). They return the first comer, so you can simply throw `Excpetion` and `catch` it above.